service/src/main/java/org/broadinstitute/listener/relay/inspectors/RequestLogger.java

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.listener.relay.inspectors;
+
+import com.microsoft.azure.relay.RelayedHttpListenerRequest;
+import java.io.IOException;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.TreeMap;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.listener.relay.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RequestLogger {
+
+  private final Logger logger = LoggerFactory.getLogger(RequestLogger.class);
+  private static final List<String> MUST_MASKED_HEADER_NAMES = List.of("Authorization", "Cookie");
+
+  /**
+   * Logs a relayed HTTP request with the result status code
+   *
+   * @param relayedHttpListenerRequest Relayed request
+   * @param statusCode Result of the request
+   * @param requestTimestamp Timestamp of the request
+   * @param prefix Logging prefix to include
+   * @throws IOException
+   * @throws InterruptedException
+   */
+  public void logRequest(
+      RelayedHttpListenerRequest relayedHttpListenerRequest,
+      int statusCode,
+      OffsetDateTime requestTimestamp,
+      String prefix)
+      throws IOException, InterruptedException {
+    if (relayedHttpListenerRequest == null) {
+      logger.warn("Null request provided for logging");
+      return;
+    }
+
+    // HTTP headers are case-insensitive
+    var headers = new TreeMap<String, String>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
This is annoying, but as rob rightly pointed out, we are open to case sensitivity issues otherwise.